src/DSE.Open.Observations.Abstractions/Observation.cs

```diff
@@ -48,7 +47,7 @@ protected Observation(ObservationId id, MeasureId measureId, long timestamp)
     [JsonInclude]
     [JsonPropertyName("t")]
     [JsonPropertyOrder(-89800)]
-    protected long Timestamp { get; }
+    public long Timestamp { get; }
```

Comment:
We're forced to expose this (unless we do internals visible to) so the source generator can access it.